sunpy/map/mapbase.py

```diff
@@ -1902,6 +1904,79 @@ def draw_limb(self, axes=None, **kwargs):
 
         return [circ]
 
+    @u.quantity_input
+    def draw_quadrangle(self, bottom_left, *, width: u.deg = None, height: u.deg = None,
+                        axes=None, top_right=None, **kwargs):
+        """
+        Draw a rectangle defined in world coordinates on the plot using Quadrangle from astropy.
+
+        This draws a rectangle that has corners at ``(bottom_left, top_right)``,
+        and has sides aligned with the coordinate axes of the frame of ``bottom_left``,
+        which may be different from the coordinate axes of the map.
+
+        If ``width`` and ``height`` are specified, they are respectively added to the
+        longitude and latitude of the ``bottom_left`` coordinate to calculate a
+        ``top_right`` coordinate.
+
+        Parameters
+        ----------
+        bottom_left : `~astropy.coordinates.SkyCoord`
+            The bottom-left coordinate of the rectangle. It can
+            have shape ``(2,)`` to simultaneously define ``top_right``.
+        top_right : `~astropy.coordinates.SkyCoord`
+            The top-right coordinate of the rectangle.
+        width : `astropy.units.Quantity`, optional
+            The width of the rectangle. Required if ``top_right`` is omitted.
+        height : `astropy.units.Quantity`
+            The height of the rectangle. Required if ``top_right`` is omitted.
+        axes : `matplotlib.axes.Axes`
+            The axes on which to plot the rectangle, defaults to the current
+            axes.
+
+        Returns
+        -------
+        rect : `list`
+            A list containing the `~astropy.visualization.wcsaxes.patches.Quadrangle` object, after
+            it has been added to ``axes``.
+
+        Notes
+        -----
+        If user's astropy version is < 4.2, raises an error to use draw_rectangle instead.
+
+        Extra keyword arguments to this function are passed through to the
+        `~astropy.visualization.wcsaxes.patches.Quadrangle` instance.
+        """
+        if version.parse(astropy.__version__) < version.parse("4.2.0"):
+            raise ImportError('Astropy >= 4.2 is required for this method, use draw_rectangle instead.')
+
+        from astropy.visualization.wcsaxes import Quadrangle
+
+        if isinstance(top_right, u.Quantity) and isinstance(width, u.Quantity):
+            # The decorator assigns the first positional arg to top_right and so on.
+            height = width
+            width = top_right
+            top_right = None
+
```

Comment:
```suggestion
```
None of this is necessary (it's vestigial in `draw_rectangle()` and should have been removed)